### PR TITLE
Fix Starlette Denial of service (DoS) via multipart/form-data

### PR DESCRIPTION
**Pull request by spolti**
chore:	Fix CVE-2024-47874

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -345,4 +345,4 @@ check-doc-links:
 	@python3 hack/verify-doc-links.py && echo "$@: OK"
 
 poetry-update-lockfiles:
-	bash -ec 'for value in $$(find . -name poetry.lock -exec dirname {} \;); do (cd "$${value}" && echo "Updating $${value}/poetry.lock" && poetry update --lock); done'
+	bash -ec 'for value in $$(find . -name poetry.lock -exec dirname {} \;); do (cd "$${value}" && echo "Updating $${value}/poetry.lock" && poetry lock --no-update); done'
```
Comment (terrytangyuan) [inline Makefile]: Could you clarify why this change is needed?

Comment (spolti) [reply Makefile]: it is not :)
reverted back

File: python/kserve/pyproject.toml [hunk 2]
```diff
@@ -35,7 +35,9 @@ exclude = ["kserve/storage/test"]
 [tool.poetry.dependencies]
 python = ">=3.9,<3.13"
 uvicorn = { version = "^0.30.6", extras = ["standard"] }
-fastapi = ">=0.114.1"
+fastapi = ">=0.115.2"
+# pin starlette version to >= 0.40.0 as fastapi supports it
+starlette = ">=0.40.0,<0.41.0"
```
Comment (yuzisun) [inline python/kserve/pyproject.toml]: why need to pin the starlette version ?

Comment (spolti) [reply python/kserve/pyproject.toml]: I kept the same range from fastapi, otherwise, it would install the old version instead of 0.40.0

Comment (sivanantha321) [reply python/kserve/pyproject.toml]: We can update the fastapi version to >=0.115.3 which defines this version range for starlette https://fastapi.tiangolo.com/release-notes/#01153

Comment (spolti) [reply python/kserve/pyproject.toml]: great :)

Comment (spolti) [reply python/kserve/pyproject.toml]: @yuzisun @sivanantha321 updated.

**Review by sivanantha321 — APPROVED** [review]

**Pull request merged**